File: articles/decision-making-under-uncertainty/cpp/decision_uncertainty_score.cpp
// Efficient decision profile scoring example.
// Compile: g++ decision_uncertainty_score.cpp -std=c++17 -O2 -o decision_uncertainty_score
// Run: ./decision_uncertainty_score

#include <algorithm>
#include <iostream>
#include <string>
#include <vector>

struct OptionProfile {
    std::string name;
    double expected_return;
    double robustness;
    double flexibility;
    double information_quality;
    double exposure;
    double option_value;
    double reversibility;
    double ethical_resilience;
    double learning_value;
};

double decision_profile(const OptionProfile& o) {
    return 0.14 * o.expected_return +
           0.18 * o.robustness +
           0.16 * o.flexibility +
           0.12 * o.information_quality -
           0.16 * o.exposure +
           0.14 * o.option_value +
           0.10 * o.reversibility +
           0.10 * o.ethical_resilience +
           0.10 * o.learning_value;
}

double fragility_risk(const OptionProfile& o) {
    return 0.24 * o.exposure +
           0.18 * (1.0 - o.robustness) +
           0.14 * (1.0 - o.flexibility) +
           0.13 * (1.0 - o.option_value) +
           0.12 * (1.0 - o.reversibility) +
           0.10 * (1.0 - o.ethical_resilience) +
           0.09 * (1.0 - o.information_quality);
}

int main() {
    std::vector<OptionProfile> options = {
        {"High-Return Brittle Option", 0.86, 0.28, 0.31, 0.63, 0.82, 0.26, 0.22, 0.38, 0.30},
        {"Balanced Robust Option", 0.72, 0.79, 0.74, 0.72, 0.44, 0.72, 0.68, 0.68, 0.70},
        {"Exploratory Optionality Option", 0.61, 0.71, 0.88, 0.49, 0.53, 0.86, 0.82, 0.62, 0.88}
    };

    std::sort(options.begin(), options.end(), [](const OptionProfile& a, const OptionProfile& b) {
        return decision_profile(a) > decision_profile(b);
    });

    for (const auto& option : options) {
        std::cout << option.name
                  << " | profile " << decision_profile(option)
                  << " | fragility " << fragility_risk(option) << "\n";
    }

    return 0;
}

File: articles/decision-making-under-uncertainty/rust/decision_uncertainty_score.rs
// Command-line decision-making under uncertainty scoring scaffold.
// Compile: rustc decision_uncertainty_score.rs -o decision_uncertainty_score
// Run: ./decision_uncertainty_score

#[derive(Debug)]
struct OptionProfile {
    name: &'static str,
    expected_return: f64,
    robustness: f64,
    flexibility: f64,
    information_quality: f64,
    exposure: f64,
    option_value: f64,
    reversibility: f64,
    ethical_resilience: f64,
    learning_value: f64,
}

fn decision_profile(o: &OptionProfile) -> f64 {
    0.14 * o.expected_return
        + 0.18 * o.robustness
        + 0.16 * o.flexibility
        + 0.12 * o.information_quality
        - 0.16 * o.exposure
        + 0.14 * o.option_value
        + 0.10 * o.reversibility
        + 0.10 * o.ethical_resilience
        + 0.10 * o.learning_value
}

fn fragility_risk(o: &OptionProfile) -> f64 {
    0.24 * o.exposure
        + 0.18 * (1.0 - o.robustness)
        + 0.14 * (1.0 - o.flexibility)
        + 0.13 * (1.0 - o.option_value)
        + 0.12 * (1.0 - o.reversibility)
        + 0.10 * (1.0 - o.ethical_resilience)
        + 0.09 * (1.0 - o.information_quality)
}

fn main() {
    let options = vec![
        OptionProfile { name: "High-Return Brittle Option", expected_return: 0.86, robustness: 0.28, flexibility: 0.31, information_quality: 0.63, exposure: 0.82, option_value: 0.26, reversibility: 0.22, ethical_resilience: 0.38, learning_value: 0.30 },
        OptionProfile { name: "Balanced Robust Option", expected_return: 0.72, robustness: 0.79, flexibility: 0.74, information_quality: 0.72, exposure: 0.44, option_value: 0.72, reversibility: 0.68, ethical_resilience: 0.68, learning_value: 0.70 },
        OptionProfile { name: "Exploratory Optionality Option", expected_return: 0.61, robustness: 0.71, flexibility: 0.88, information_quality: 0.49, exposure: 0.53, option_value: 0.86, reversibility: 0.82, ethical_resilience: 0.62, learning_value: 0.88 },
    ];

    for o in options {
        println!(
            "{} | profile {:.3} | fragility {:.3}",
            o.name,
            decision_profile(&o),
            fragility_risk(&o)
        );
    }
}
